query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -12,16 +16,82 @@ use std::{
 use tracing::log::trace;
 use user_facing_errors::query_engine::DatabaseConstraint;
 
+async fn generate_id(
+    conn: &dyn QueryExt,
+    conn_info: ConnectionInfo,
+    model: &ModelRef,
+    args: WriteArgs,
+    trace_id: Option<String>,
+) -> crate::Result<(Option<SelectionResult>, bool)> {
+    let pk = model.primary_identifier();
+
+    // fast path if it's not mysql
+    if conn_info.sql_family() != SqlFamily::Mysql {
+        return Ok((args.as_record_projection(pk.into()), false));
+    }
+
+    // Go to all the values and generate a select statement with the correct mysql function
+    let (pk_select, generated) = pk
```

Comment:
Could you move this into its own function. Its pretty complicated section of code.

Reply:
Not really, there are reasons why everything is in a single function, this functionality used to be together with `write::create_record` in a single function. I really feel what is doing right now is atomic itself and just relevant if the engine is MySQL. Now, what is true is that probably I could pull out that `if` outside this function. Let me see if I can figure something out.